src/Transport/Sending/IBatcher.cs

```diff
@@ -1,10 +1,11 @@
 namespace NServiceBus.Transport.AzureServiceBus
 {
     using System.Collections.Generic;
+    using System.Threading.Tasks;
     using Transport;
 
     interface IBatcherInternal
     {
-        IList<BatchInternal> ToBatches(TransportOperations operations);
+        Task<List<BatchInternal>> ToBatches(TransportOperations operations);
```

Comment:
@yvesgoeleven that's what the async local does there. Agree it would be better to not return a task here. I think we have two options:

- Address it now (I'm very limited with my availability so @yvesgoeleven would need to jump in)
- Merge it with a TODO and address it in the larger PDev refactorings (we basically build dept now)

I'm OK with both approaches. If you need/want my help it would have to wait until my return (not implying I'm required here just offering my help)